lib/instrument/transformer/nskeyed.js

```diff
@@ -0,0 +1,389 @@
+      log.error(`unsupported encoder: ${plist.$archiver}`);
+      return false;
+    }
+    if (plist.$version !== 100_000) {
+      log.error(`expected ${NSKeyedArchiveVersion}, got ${plist.$version }`);
+      return false;
+    }
+    const top = plist.$top;
+    const topUID = top.root;
+    if (!topUID) {
+      log.error(`top object did not have a UID! dump: ${top}`);
+      return false;
+    }
+    this.topUID = topUID;
+    this.objects = plist.$objects;
+  }
+
+  /**
+   * use the UNARCHIVE_CLASS_MAP to find the unarchiving delegate of a uid
+   */
+  classForUid (index) {
+    const meta = this.objects[index.UID];
+    const name = meta.$classname;
+    const klass = UNARCHIVE_CLASS_MAP[name];
+    if (!klass) {
+      log.error(`no mapping for ${name} in UNARCHIVE_CLASS_MAP`);
+    }
+    return klass;
+  }
+  decodeKey (obj, key) {
+    const val = obj[key];
+    if (typeof val === 'object' && val.UID) {
+      return this.decodeObject(val);
+    }
+    return val;
+  }
+
+  decodeObject (index) {
+    if (index === NULL_UID) {
+      return null;
+    }
+    const obj = this.unpackedUids[index];
+    if (obj === CYCLE_TOKEN) {
+      log.error(`archive has a cycle with ${index}`);
+      return null;
+    }
+    if (typeof obj !== 'undefined') {
+      return obj;
+    }
+    const rawObj = this.objects[index.UID];
+    this.unpackedUids[index.UID] = CYCLE_TOKEN;
+
+    if (typeof rawObj !== 'object' || !rawObj.$class) {
+      this.unpackedUids[index.UID] = obj;
+      return rawObj;
+    }
+    const classUid = rawObj.$class;
+    if (!classUid) {
+      log.error(`object has no $class: ${rawObj}`);
+      return null;
+    }
+    const klass = this.classForUid(classUid);
+    const klassObj = new klass().decodeArchive(new ArchivedObject(rawObj, this));
+    this.unpackedUids[index.UID] = klassObj;
+    return klassObj;
+  }
+  toObject () {
+    this.unpackArchiveHeader();
+    return this.decodeObject(this.topUID);
+  }
+}
+
+
+/**
+ * Capable of packing an object tree into the NSKeyedArchive format.
+ * Apple's implementation can be found here:
+ * https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSKeyedArchiver.swift
+ */
+class Archive {
+  constructor (inputObject) {
+    this.input = inputObject;
+    this.classMap = {};
+    this.objects = ['$null'];
+  }
+
+  uidForArchiver (archiver) {
+    let val = this.classMap[archiver];
+    if (val) {
+      return new plistlib.UID(val);
+    }
+    this.classMap[archiver] = this.objects.length;
+    val = new plistlib.UID(this.objects.length);
+    this.objects.push({
+      '$classes': [archiver],
+      '$classname': archiver
+    });
+    return val;
+  }
+  archive (obj) {
+    if (typeof obj === 'undefined') {
+      return NULL_UID;
+    }
+    const index = new plistlib.UID(this.objects.length);
+    const objType = typeof obj;
+    if (['number', 'string', 'boolean', 'UID'].indexOf(objType) !== -1) {
```

Comment:
- this list could be extracted to a constant
- consider using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes instead of indexOf